lib/sinon/collection.js

```diff
@@ -91,46 +90,22 @@ var collection = {
         return this.add(sinonSpy.apply(sinonSpy, arguments));
     },
 
-    stub: function stub(object, property, value) {
-        if (property) {
-            if (!object) {
-                var type = object === null ? "null" : "undefined";
-                throw new Error("Trying to stub property '" + valueToString(property) + "' of " + type);
-            }
+    stub: function stub(object, property/*, value*/) {
+        failOnFalsyObject.apply(null, arguments);
 
-            var original = object[property];
+        var isStubbingEntireObject = typeof property === "undefined" && typeof object === "object";
+        var isStubbingNonFunctionProperty = property && typeof object[property] !== "function";
+        var stubbed = isStubbingNonFunctionProperty ?
+                        stubNonFunctionProperty.apply(null, arguments) :
+                        sinonStub.apply(null, arguments);
 
-            if (typeof original !== "function") {
-                if (!hasOwnProperty.call(object, property)) {
-                    throw new TypeError("Cannot stub non-existent own property " + valueToString(property));
-                }
-
-                object[property] = value;
-
-                return this.add({
-                    restore: function () {
-                        object[property] = original;
-                    }
-                });
-            }
-        }
-        if (!property && !!object && typeof object === "object") {
-            var col = this;
-            var stubbedObj = sinonStub.apply(null, arguments);
-
-            walk(stubbedObj, function (prop, propOwner) {
-                if (
-                    typeof getPropertyDescriptor(propOwner, prop).value === "function" &&
-                    stubbedObj.hasOwnProperty(prop)
-                ) {
-                    col.add(stubbedObj[prop]);
-                }
-            });
-
-            return stubbedObj;
+        if (isStubbingEntireObject) {
+            collectOwnMethods(stubbed).forEach(this.add.bind(this));
```

Comment:
I did, don't use it much, as I try to avoid using `this` as I find it reduces problems when I do :)